saw-core/src/SAWCore/Rewriter.hs

```diff
@@ -233,15 +252,14 @@ scMatch sc pat term =
     -- Check if a term is a higher-order variable pattern, i.e., a free variable
     -- (meaning one that can match anything) applied to 0 or more bound variable
     -- arguments. Depth is the number of variables bound by lambdas or pis since
-    -- the top of the current pattern, so "free" means >= the current depth and
-    -- "bound" means less than the current depth
-    asVarPat :: Int -> Term -> Maybe (DeBruijnIndex, [DeBruijnIndex])
+    -- the top of the current pattern, so "bound" means less than the current depth
+    asVarPat :: Int -> Term -> Maybe (VarIndex, [DeBruijnIndex])
     asVarPat depth = go []
       where
         go js x =
           case unwrapTermF x of
-            LocalVar i
-              | i >= depth -> Just (i, js)
+            FTermF (ExtCns ec)
+              | IntSet.member (ecVarIndex ec) ixs -> Just (ecVarIndex ec, js)
```

Comment:
...as well as here. Perhaps we should just pass the context around as an `IntSet` instead of a list?...